src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -454,6 +454,20 @@ void primitive_inst::update_shape() {
         if (var_mem_size < _impl_params->get_output_layout(0).get_buffer_size().count())
             set_shape_change();
     }
+
+#ifdef ENABLE_ONEDNN_FOR_GPU
+    if (get_node().is_type<fully_connected>() && get_node().get_preferred_impl_type() == impl_types::onednn) {
+        auto attrs_onednn = std::make_shared<dnnl::primitive_attr>();
+        std::vector<cldnn::fused_primitive_desc_onednn> fused_desc_onednn;
+        cldnn::create_onednn_primitive_attributes(get_node(),
+                                                    _impl_params->fused_desc,
+                                                    attrs_onednn,
+                                                    fused_desc_onednn,
+                                                    _impl_params.get());
+        _impl_params->attrs_onednn = attrs_onednn;
+        _impl_params->fused_desc_onednn = fused_desc_onednn;
+    }
```

Comment:
1) Why don't we moving this to update_impl so that it will be called only when the shape is changed? 
2) Also you already limited fusion type to fc in the prepare_primitive_fusing. So I think no need to limit it to fc here again.

Reply:
moved it.